src/addrman.cpp

```diff
@@ -459,11 +459,14 @@ CAddrInfo CAddrMan::Select_(bool newOnly)
     }
 }
 
-#ifdef DEBUG_ADDRMAN
 int CAddrMan::Check_()
 {
     AssertLockHeld(cs);
 
+    // Run consistency checks 1 in m_consistency_check_ratio times if enabled
+    if (m_consistency_check_ratio == 0) return 0;
+    if (GetRand(m_consistency_check_ratio) >= 1) return 0;
```

Comment:
Perhaps use `insecure_rand.randrange(m_consistency_check_ratio)`? Probably orders of magnitude faster...

Reply:
See previous discussion: https://github.com/bitcoin/bitcoin/pull/20233#discussion_r672913835

Reply:
I think I was mistaken about this causing problems for the unit tests. I'll try again with insecure_rand tomorrow.

Reply:
Done.